app/models/manageiq/providers/base_manager/metrics_capture.rb

```diff
@@ -175,18 +175,41 @@ def create_rollup_task_for_cluster(ems_cluster, hosts)
     end
   end
 
+  # number of records to send at a time for a single capture message
+  # @return [Numeric] 0 for no batching.  (defaults to 0)
+  def batch_size
+    @batch_size ||= ::Settings.dig(:ems, "ems_#{ems.provider_name.underscore}", :capture_batch_size).to_i
+  end
+
   def perf_capture_queue_targets(targets, interval, start_time:, end_time:, parent: nil)
-    task = create_rollup_task_for_cluster(parent, targets) if parent
+    if batch_size > 1
+      if parent # send all of the hosts in a cluster in one message
+        perf_capture_queue_target(targets, interval, :start_time => start_time, :end_time => end_time, :rollup => true)
+      else # batch sends
+        targets.each_slice(batch_size) do |targets_batch|
+          perf_capture_queue_target(targets_batch, interval, :start_time => start_time, :end_time => end_time)
+        end
+      end
+    else # send individual messages
+      task = create_rollup_task_for_cluster(parent, targets) if parent
+      perf_capture_queue_targets_individ(targets, interval, :start_time => start_time, :end_time => end_time, :task_id => task&.id)
+    end
+  end
+
+  # standard one message per object to collect
+  def perf_capture_queue_targets_individ(targets, interval, start_time:, end_time:, task_id: nil)
```

Comment:
Not sure on the method name here (I'm usually adverse to abbrevs).

Reply:
I put this in to allow sub classing to override this stuff.
Maybe that is no longer necessary